File: regex_extracting/extracting/regex_lib/CameraFeature.py
# encoding=utf8
"""
    Author:  'jdwang'
    Date:    'create date: 2017-01-13'; 'last updated date: 2017-01-13'
    Email:   '383287471@qq.com'
    Describe:
"""
from __future__ import print_function
from regex_extracting.extracting.common.regex_base import RegexBase

__version__ = '1.3'


class CameraFeature(RegexBase):
    name = '拍照功能'

    def __init__(self, sentence):
        # 要处理的输入句子

        self.sentence = sentence

        # region 1 初始化正则表达式
        # 描述正则表达式
        self.statement_regexs = [
            '拍照功能|照相功能|拍摄功能'
        ]
        # 值正则表达式
        self.value_regexs = [
            '好友照片分享|脸部识别功能|人脸幻灯片|人脸识别',
            '随意|随便|都可以|其他|别的',
        ]
        # endregion
        super(CameraFeature, self).__int__()
        self.regex_process()


if __name__ == '__main__':
    price = CameraFeature('价不高')
    for info_meta_data in price.info_meta_data_list:
        print('-' * 80)
        print(str(info_meta_data))

File: regex_extracting/asp_version/Extracting/CameraFeature.cs
/*
 * author : wxp
 * date : 2013/9/11
 * */

using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;
using System.Web;

namespace Ch2R.Code.Extracting
{
    public class CameraFeature : ParameterBase
    {
        public CameraFeature(Areas.Chat.Models.UserSession userSession, string sentence)
            : base(userSession, sentence)
        {
            name = "拍照功能";

            #region 初始化正则表达式
            // 描述正则表达式
            StatementRegexs = new string[]
            {
                @"拍照功能|照相功能|拍摄功能"    
            };
            // 值正则表达式
            ValueRegexs = new string[]
            {
                @"好友照片分享|脸部识别功能|人脸幻灯片|人脸识别",
                @"随意|随便|都可以|其他|别的"
            };
            #endregion

            RegexProcess();
        }

        public override void Format()
        {
            for (int i = 0; i < list.Count; ++i)
            {
                if (Regex.IsMatch(list[i].rawData, @"随意|随便|都可以"))
                {
                    list[i].normalData = "";
                }
                else if (Regex.IsMatch(list[i].rawData, @"别的|其他的"))
                {
                    if (userSession.liveTable.Parameter["拍照功能"] != null)
                    {
                        list[i].normalData = (String)userSession.liveTable.Parameter["拍照功能"];
                        list[i].isNegative = true;
                    }
                    else
                    {
                        /**
                         * 仍有其他价格为null的情况
                         * 别的未测试
                         * */
                    }
                }
                else
                {
                    list[i].normalData = list[i].rawData;
                }
                /* 一般不会有人说明不要某种拍照模式的
                 * 所以在此不做否定判断
                 */
            }
        }

        public override InfoBlock ToInfoBlock()
        {
            List<InfoUnit> L = new List<InfoUnit>();
            foreach (InfoMetadata im in list)
            {
                InfoUnit iu = new InfoUnit();
                iu.normalData = im.normalData;
                iu.rawData = im.rawData;
                iu.isNegative = im.isNegative;
                L.Add(iu);
            }
            InfoBlock ib = new InfoBlock(name, "and", L);
            /* 这里关系应该是“and”
             * 用户对拍照的功能可以是多个并存的
             */
            return ib;
        }
    }
}